### PR TITLE
Enable null warnings and eliminate the existing ones

### LINKED ISSUE #64: .NET: Invokes may pass null to non-nullable args
`Deployment.InvokeAsync` [expects](https://github.com/pulumi/pulumi/blob/master/sdk/dotnet/Pulumi/Deployment/Deployment_Invoke.cs#L20) a non-nullable `ResourceArgs args`. Meanwhile, TF Gen sometimes generates code that passes null in there: [example](https://github.com/pulumi/pulumi-azure/blob/master/sdk/dotnet/Core/GetClientConfig.cs#L18), [source](https://github.com/pulumi/pulumi-terraform-bridge/blob/7225dfb0ac605582028e32ed7d6cf9ab1db120e1/pkg/tfgen/generate_csharp.go#L850). This leads to a null reference exception when `Invoke` is called.

Node SDK [passes](https://github.com/pulumi/pulumi-azure/blob/master/sdk/nodejs/core/getClientConfig.ts#L31-L32) an empty object in the same case. So, I suggest doing the same for .NET - pass `ResourceArgs.Empty`. 

Bonus question: why doesn't `null` yield a warning at compile time?

cc @pgavlin

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Fixes #64

### CODE REVIEW

File: pkg/tfgen/generate_csharp_templates.go [hunk 1]
```diff
@@ -73,9 +73,12 @@ namespace {{.Namespace}}
         {
             var assembly = typeof(Utilities).GetTypeInfo().Assembly;
             using (var stream = assembly.GetManifestResourceStream("{{.Namespace}}.version.txt"))
```
Comment (CyrusNajmabadi) [inline pkg/tfgen/generate_csharp_templates.go]:
```suggestion
            using var stream = assembly.GetManifestResourceStream("{{.Namespace}}.version.txt");
```

File: pkg/tfgen/generate_csharp_templates.go [hunk 2]
```diff
@@ -73,9 +73,12 @@ namespace {{.Namespace}}
...
-            using (var reader = new StreamReader(stream))
             {
-                version = reader.ReadToEnd().Trim();
+                if (stream == null) throw new Exception("Embedded version.txt file failed to load");
+                using (var reader = new StreamReader(stream))
```
Comment (CyrusNajmabadi) [inline pkg/tfgen/generate_csharp_templates.go]:
```suggestion
                using var reader = new StreamReader(stream ?? throw new UnsupportedOperationException(...));
```

**Comment by mikhailshilkov** [comment]
I don't have permissions to merge this one, @CyrusNajmabadi could you please?

**Comment by CyrusNajmabadi** [comment]
Done!

**Comment by mikhailshilkov** [comment]
@CyrusNajmabadi Thanks! A copy is here https://github.com/pulumi/pulumi-terraform/pull/514

**Pull request merged**